src/utils/formatRequestsData.ts

```diff
@@ -10,30 +10,34 @@ import {
   getDispute,
   getStatus,
   isFinalResponse,
+  TimeStamps,
 } from '~/utils';
 
 export const formatRequestsData = (
   requestsFullData: RequestFullData[],
   ensNames: EnsNames,
   metadatas: Metadata[],
+  timestamps: TimeStamps[],
```

Comment:
I will upload a minor update of the returned data not indexed with numbers but with keys so the mappings can be avoided and it easier the access:

```
private mapEventArgsToRequestWithId(eventArgs: any[]): RequestWithId {
    return {
      requestId: eventArgs[0],
      request: {
        nonce: eventArgs[1][0],
        requester: eventArgs[1][1],
        requestModule: eventArgs[1][2],
        responseModule: eventArgs[1][3],
        disputeModule: eventArgs[1][4],
        resolutionModule: eventArgs[1][5],
        finalityModule: eventArgs[1][6],
        requestModuleData: eventArgs[1][7],
        responseModuleData: eventArgs[1][8],
        disputeModuleData: eventArgs[1][9],
        resolutionModuleData: eventArgs[1][10],
        finalityModuleData: eventArgs[1][11],
      },
      ipfsHash: eventArgs[2],
      blockNumber: eventArgs[3],
    };
  }
  ```

Reply:
done! https://github.com/defi-wonderland/prophet-app/pull/1/commits/2c39d568d8bbf61818d4c32a8f0748b7e7dd8f15